### PR TITLE
Ginkgo removal branch merged into main.

### PR DESCRIPTION
**Pull request by greyerof**
This PR merges the ginkgo_removal branch into main.

**IMPORTANT**: It must me submitted using the "Create merge commit" option, in order to preserve the history (git blame) of every change in both branches.

### CODE REVIEW

File: .github/workflows/pre-main.yaml [hunk 1]
```diff
@@ -249,28 +249,28 @@ jobs:
...
+
+      # - name: Check that their are 0 failed tests in the gradetool results
+      #   run: |
+      #     if $(jq '.[] | .Fail | length' "${GITHUB_WORKSPACE}"/results.txt | grep -q 0); then
+      #       echo OK
+      #     else
+      #       echo "Gradetool has found failing tests in the following:"
+      #       jq '.[] | .Fail | .[].id' "${GITHUB_WORKSPACE}"/results.txt
+      #       exit 1
+      #     fi
```
Comment (sebrandon1) [inline .github/workflows/pre-main.yaml:273]: We can uncomment this in a subsequent `main` branch PR.

**Review by sebrandon1 — APPROVED** [review]
Waiting for all green!  ✅

File: .github/workflows/pre-main.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ on:
   push:
     branches: [ main ]
   pull_request:
-    branches: [ main ]
+    branches: [ main, ginkgo_removal]
```
Comment (jmontesi) [inline .github/workflows/pre-main.yaml:7]: The references to the `ginkgo_removal` branch should be deleted.

Comment (sebrandon1) [reply .github/workflows/pre-main.yaml:7]: We'll have to do that in a subsequent PR as well.  Same thing in the QE PR.

**Comment by rdavid** [comment]
All green!

**Review by rdavid — APPROVED** [review]

**Comment by sebrandon1** [comment]
Marked `do-not-merge` until @greyerof is ready.  Remember, we have to merge with the 'Create a merge commit' option, not the squash option.

**Review by ramperher — APPROVED** [review]
Looks fine for us too. By the way, does this mean that ginkgo_removal branch will be removed? Just to take it into account too in our automation.

**Comment by greyerof** [comment]
> Looks fine for us too. By the way, does this mean that ginkgo_removal branch will be removed? Just to take it into account too in our automation.

@ramperher it won't be used anymore but we can keep it as long as needed.

**Review by jmontesi — APPROVED** [review]

**Pull request merged**